scripts/compare_benchmarks.py

```diff
@@ -84,74 +144,194 @@ def calculate_and_format_p_value(old, new):
+# spanning two columns
 table_data = []
-table_data.append(["Benchmark", "prev. iter/s", "runs", "new iter/s", "runs", "change [%]", "p-value"])
+table_data.append(["Item", "$latency", "", "Change", "$thrghpt", "", "Change", "p-value"])
+table_data.append(["", "old", "new", "", "old", "new", "", ""])
 
 for old, new in zip(old_data["benchmarks"], new_data["benchmarks"]):
     name = old["name"]
     if old["name"] != new["name"]:
         name += " -> " + new["name"]
 
-    if float(old["items_per_second"]) > 0.0:
-        diff = float(new["items_per_second"]) / float(old["items_per_second"])
-        diffs.append(diff)
+    # Create numpy arrays for old/new successfull/unsuccessful runs from benchmark dictionary
+    old_successful_durations = np.array([run["duration"] for run in old["successful_runs"]], dtype=np.float64)
+    new_successful_durations = np.array([run["duration"] for run in new["successful_runs"]], dtype=np.float64)
+    old_unsuccessful_durations = np.array([run["duration"] for run in old["unsuccessful_runs"]], dtype=np.float64)
+    new_unsuccessful_durations = np.array([run["duration"] for run in new["unsuccessful_runs"]], dtype=np.float64)
+    old_avg_successful_duration = np.mean(old_successful_durations)  # defaults to np.float64 for int input
+    new_avg_successful_duration = np.mean(new_successful_durations)
+
+    total_runtime_old += old_avg_successful_duration if not math.isnan(old_avg_successful_duration) else 0.0
+    total_runtime_new += new_avg_successful_duration if not math.isnan(new_avg_successful_duration) else 0.0
+
+    # Check for duration==0 to avoid div/0
+    if float(old_avg_successful_duration) > 0.0:
+        diff_duration = float(new_avg_successful_duration / old_avg_successful_duration)
     else:
-        diff = float("nan")
+        diff_duration = float("nan")
 
-    diff_formatted = format_diff(diff)
-    p_value_formatted = calculate_and_format_p_value(old, new)
+    if float(old["items_per_second"]) > 0.0:
+        diff_throughput = float(new["items_per_second"]) / float(old["items_per_second"])
+        diffs_throughput.append(diff_throughput)
+    else:
+        diff_throughput = float("nan")
+
+    # Format the diffs (add colors and percentage output) and calculate p-value
+    diff_throughput_formatted = color_diff(diff_throughput)
+    diff_duration_formatted = color_diff(diff_duration, True)
+    p_value_formatted = calculate_and_format_p_value(old_successful_durations, new_successful_durations)
+
+    old_iteration_count = len(old_successful_durations) + len(old_unsuccessful_durations)
+    new_iteration_count = len(new_successful_durations) + len(new_unsuccessful_durations)
+    if (old_data["context"]["max_runs"] > 0 or new_data["context"]["max_runs"] > 0) and (
+        old_iteration_count >= old_data["context"]["max_runs"] or new_iteration_count >= new_data["context"]["max_runs"]
```

Comment:
Why `>=`? If it was capped, shouldn't `>` be impossible (and lead to a warning/failure)?

Reply:
Changed to `<` as lower runs mean that the runtime stopped the item and equal values "should" mean that it was capped.
Thanks for the catch.